### PR TITLE
Add Console#mapK

### PR DESCRIPTION
**Pull request by kubukoz**
Initial version, if it looks nice I'll add some tests :)

### CODE REVIEW

**Comment by gvolpe** [comment]
Go for the tests, looks great, thanks!

**Comment by kubukoz** [comment]
Please take a look in a spare while :)

**Comment by kubukoz** [comment]
Also note that this is binary breaking for <2.12, which I'm not sure you're concerned about below 1.0

**Comment by gvolpe** [comment]
Thanks, will look into details tomorrow, gotta leave now (actually on small holidays :) )

**Comment by kubukoz** [comment]
Sure, have fun!

File: core/src/test/scala/cats/effect/ConsoleSpec.scala [hunk 1]
```diff
@@ -16,49 +16,50 @@
...
+    override def putStrLn[A: Show](a: A): F[Unit] =
       out1.update(acc => acc ::: a.show :: Nil)
-    override def putStr[A: Show](a: A): IO[Unit] =
+    override def putStr[A: Show](a: A): F[Unit] =
       out2.update(acc => acc ::: a.show :: Nil)
-    override def putError[A: Show](a: A): IO[Unit] =
+    override def putError[A: Show](a: A): F[Unit] =
       out3.update(acc => acc ::: a.show :: Nil)
-    override def readLn: IO[String] = IO.pure("test")
+    override def readLn: F[String] = Applicative[F].pure("test")
```
Comment (gvolpe) [inline core/src/test/scala/cats/effect/ConsoleSpec.scala]: Nitpick: since we are already import all the syntax I'd rather do `"test".pure[F]` :)

**Pull request merged**